configure.ac

```diff
@@ -4682,6 +4682,76 @@ if test "$with_libpython" != "xno"; then
 fi
 # }}} --with-libpython
 
+# --with-libpython3 {{{
+AC_ARG_VAR([LIBPYTHON3_CPPFLAGS], [Preprocessor flags for libpython3])
+AC_ARG_VAR([LIBPYTHON3_LDFLAGS], [Linker flags for libpython3])
+AC_ARG_VAR([LIBPYTHON3_LIBS], [Libraries for libpython3])
+
+AC_ARG_WITH([libpython3],
+  [AS_HELP_STRING([--with-libpython3], [if we should build with libpython3 @<:@default=yes@:>@])],
+  [with_libpython3="$withval"],
+  [with_libpython3="check"]
+)
+
+if test "$with_libpython3" != "no"; then
+  if test "$LIBPYTHON3_CPPFLAGS" = "" && test "$LIBPYTHON3_LDFLAGS" = ""; then
+    AC_ARG_VAR([PYTHON3_CONFIG], [path to python3-config])
+    AC_PATH_PROGS([PYTHON3_CONFIG],
+      [python3-config]
+    )
+    if test "$PYTHON3_CONFIG" = ""; then
+      if test "$with_libpython3" = "yes"; then
+        AC_MSG_ERROR([Unable to find python3-config])
+      fi
+      with_libpython3="no"
+    fi
+  fi
+fi
+
+if test "$PYTHON3_CONFIG" != ""; then
+  LIBPYTHON3_CPPFLAGS="`${PYTHON3_CONFIG} --includes`"
+  if test $? -ne 0; then
+    with_libpython3="no"
+  fi
+  LIBPYTHON3_LDFLAGS="`${PYTHON3_CONFIG} --ldflags`"
+  if test $? -ne 0; then
+    with_libpython3="no"
+  fi
+  LIBPYTHON3_LIBS="`${PYTHON3_CONFIG} --libs`"
+  if test $? -ne 0; then
+    with_libpython3="no"
+  fi
+fi
+
+if test "$with_libpython3" != "xno"; then
```

Comment:
What does `xno` mean? I see there's a spot where we do the comparison `if test "x$with_libpython3" != "xno"` with the `x$with_libpython3`, was that the intention here?

Reply:
Great catch. This was copied from the `libpython` tests, and there it looks like a zero-day bug, introduced when those checks were written (in `#1851`). Fixed in both places.